// @heygen/streaming-avatar/dist/models/InterruptResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Avatar SDK
 * Heygen Streaming Avatar
 *
 * The version of the OpenAPI document: 1.0.4
 * Contact: api@heygen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InterruptResponseToJSON = exports.InterruptResponseFromJSONTyped = exports.InterruptResponseFromJSON = exports.instanceOfInterruptResponse = void 0;
/**
 * Check if a given object implements the InterruptResponse interface.
 */
function instanceOfInterruptResponse(value) {
    return true;
}
exports.instanceOfInterruptResponse = instanceOfInterruptResponse;
function InterruptResponseFromJSON(json) {
    return InterruptResponseFromJSONTyped(json, false);
}
exports.InterruptResponseFromJSON = InterruptResponseFromJSON;
function InterruptResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'code': json['code'] == null ? undefined : json['code'],
        'message': json['message'] == null ? undefined : json['message'],
    };
}
exports.InterruptResponseFromJSONTyped = InterruptResponseFromJSONTyped;
function InterruptResponseToJSON(value) {
    if (value == null) {
        return value;
    }
    return {
        'code': value['code'],
        'message': value['message'],
    };
}
exports.InterruptResponseToJSON = InterruptResponseToJSON;
