// @heygen/streaming-avatar/dist/models/InterruptRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Avatar SDK
 * Heygen Streaming Avatar
 *
 * The version of the OpenAPI document: 1.0.4
 * Contact: api@heygen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InterruptRequestToJSON = exports.InterruptRequestFromJSONTyped = exports.InterruptRequestFromJSON = exports.instanceOfInterruptRequest = void 0;
/**
 * Check if a given object implements the InterruptRequest interface.
 */
function instanceOfInterruptRequest(value) {
    return true;
}
exports.instanceOfInterruptRequest = instanceOfInterruptRequest;
function InterruptRequestFromJSON(json) {
    return InterruptRequestFromJSONTyped(json, false);
}
exports.InterruptRequestFromJSON = InterruptRequestFromJSON;
function InterruptRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'sessionId': json['session_id'] == null ? undefined : json['session_id'],
    };
}
exports.InterruptRequestFromJSONTyped = InterruptRequestFromJSONTyped;
function InterruptRequestToJSON(value) {
    if (value == null) {
        return value;
    }
    return {
        'session_id': value['sessionId'],
    };
}
exports.InterruptRequestToJSON = InterruptRequestToJSON;
