// @heygen/streaming-avatar/dist/models/InterruptResponseData.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Avatar SDK
 * Heygen Streaming Avatar
 *
 * The version of the OpenAPI document: 1.0.4
 * Contact: api@heygen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InterruptResponseDataToJSON = exports.InterruptResponseDataFromJSONTyped = exports.InterruptResponseDataFromJSON = exports.instanceOfInterruptResponseData = void 0;
/**
 * Check if a given object implements the InterruptResponseData interface.
 */
function instanceOfInterruptResponseData(value) {
    return true;
}
exports.instanceOfInterruptResponseData = instanceOfInterruptResponseData;
function InterruptResponseDataFromJSON(json) {
    return InterruptResponseDataFromJSONTyped(json, false);
}
exports.InterruptResponseDataFromJSON = InterruptResponseDataFromJSON;
function InterruptResponseDataFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'code': json['code'] == null ? undefined : json['code'],
        'message': json['message'] == null ? undefined : json['message'],
    };
}
exports.InterruptResponseDataFromJSONTyped = InterruptResponseDataFromJSONTyped;
function InterruptResponseDataToJSON(value) {
    if (value == null) {
        return value;
    }
    return {
        'code': value['code'],
        'message': value['message'],
    };
}
exports.InterruptResponseDataToJSON = InterruptResponseDataToJSON;
