// @heygen/streaming-avatar/dist/models/NewSessionRequestVoice.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Avatar SDK
 * Heygen Streaming Avatar
 *
 * The version of the OpenAPI document: 1.0.4
 * Contact: api@heygen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NewSessionRequestVoiceToJSON = exports.NewSessionRequestVoiceFromJSONTyped = exports.NewSessionRequestVoiceFromJSON = exports.instanceOfNewSessionRequestVoice = exports.NewSessionRequestVoiceEmotionEnum = void 0;
/**
 * @export
 */
exports.NewSessionRequestVoiceEmotionEnum = {
    Excited: 'Excited',
    Serious: 'Serious',
    Friendly: 'Friendly',
    Soothing: 'Soothing',
    Broadcaster: 'Broadcaster'
};
/**
 * Check if a given object implements the NewSessionRequestVoice interface.
 */
function instanceOfNewSessionRequestVoice(value) {
    return true;
}
exports.instanceOfNewSessionRequestVoice = instanceOfNewSessionRequestVoice;
function NewSessionRequestVoiceFromJSON(json) {
    return NewSessionRequestVoiceFromJSONTyped(json, false);
}
exports.NewSessionRequestVoiceFromJSON = NewSessionRequestVoiceFromJSON;
function NewSessionRequestVoiceFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'voiceId': json['voice_id'] == null ? undefined : json['voice_id'],
        'rate': json['rate'] == null ? undefined : json['rate'],
        'emotion': json['emotion'] == null ? undefined : json['emotion'],
    };
}
exports.NewSessionRequestVoiceFromJSONTyped = NewSessionRequestVoiceFromJSONTyped;
function NewSessionRequestVoiceToJSON(value) {
    if (value == null) {
        return value;
    }
    return {
        'voice_id': value['voiceId'],
        'rate': value['rate'],
        'emotion': value['emotion'],
    };
}
exports.NewSessionRequestVoiceToJSON = NewSessionRequestVoiceToJSON;
