// @heygen/streaming-avatar/dist/src/models/IceResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Avatar SDK
 * Heygen Streaming Avatar
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: api@heygen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IceResponseToJSON = exports.IceResponseFromJSONTyped = exports.IceResponseFromJSON = exports.instanceOfIceResponse = void 0;
/**
 * Check if a given object implements the IceResponse interface.
 */
function instanceOfIceResponse(value) {
    return true;
}
exports.instanceOfIceResponse = instanceOfIceResponse;
function IceResponseFromJSON(json) {
    return IceResponseFromJSONTyped(json, false);
}
exports.IceResponseFromJSON = IceResponseFromJSON;
function IceResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'code': json['code'] == null ? undefined : json['code'],
        'message': json['message'] == null ? undefined : json['message'],
    };
}
exports.IceResponseFromJSONTyped = IceResponseFromJSONTyped;
function IceResponseToJSON(value) {
    if (value == null) {
        return value;
    }
    return {
        'code': value['code'],
        'message': value['message'],
    };
}
exports.IceResponseToJSON = IceResponseToJSON;
