// @heygen/streaming-avatar/dist/src/models/StopSessionResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Avatar SDK
 * Heygen Streaming Avatar
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: api@heygen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StopSessionResponseToJSON = exports.StopSessionResponseFromJSONTyped = exports.StopSessionResponseFromJSON = exports.instanceOfStopSessionResponse = void 0;
/**
 * Check if a given object implements the StopSessionResponse interface.
 */
function instanceOfStopSessionResponse(value) {
    return true;
}
exports.instanceOfStopSessionResponse = instanceOfStopSessionResponse;
function StopSessionResponseFromJSON(json) {
    return StopSessionResponseFromJSONTyped(json, false);
}
exports.StopSessionResponseFromJSON = StopSessionResponseFromJSON;
function StopSessionResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'code': json['code'] == null ? undefined : json['code'],
        'message': json['message'] == null ? undefined : json['message'],
    };
}
exports.StopSessionResponseFromJSONTyped = StopSessionResponseFromJSONTyped;
function StopSessionResponseToJSON(value) {
    if (value == null) {
        return value;
    }
    return {
        'code': value['code'],
        'message': value['message'],
    };
}
exports.StopSessionResponseToJSON = StopSessionResponseToJSON;
